src/implementation.cpp

```diff
@@ -83,10 +85,19 @@ class unsupported_implementation final : public implementation {
   ) const noexcept final {
     return UNSUPPORTED_ARCHITECTURE;
   }
-  WARN_UNUSED error_code minify(const uint8_t *, size_t, uint8_t *, size_t &) const noexcept final {
+  WARN_UNUSED error_code minify(const uint8_t *, size_t, uint8_t *, size_t &) const noexcept final override {
     return UNSUPPORTED_ARCHITECTURE;
   }
-
+  WARN_UNUSED bool validate_utf8(const char *, size_t) const noexcept final override {
```

Comment:
At this point, we should remove the unsupported implementation and issue an abort or return nullptr when there is no matching implementation. (Not this PR.) If you compile without fallback, you're kind of asking for it.

Reply:
Agreed.